docker/indexer/storage/storage.go

```diff
@@ -18,22 +18,82 @@ package storage
 
 import (
 	"context"
+	"fmt"
+	"sync"
+	"time"
 
+	"cloud.google.com/go/datastore"
 	"github.com/go-git/go-git/v5/plumbing"
 	"github.com/google/osv.dev/docker/indexer/stages/preparation"
 	"github.com/google/osv.dev/docker/indexer/stages/processing"
+)
 
-	log "github.com/golang/glog"
+const (
+	kind   = "source-repository"
+	keyFmt = "%s-%x"
 )
 
+// document represents a single repository entry in datastore.
+type document struct {
+	Name        string
+	BaseCPE     string
+	Version     string
+	Commit      []byte
+	When        time.Time
+	RepoType    string
+	FileExts    []string
+	FileResults []processing.FileResult
```

Comment:
Thinking more on extension: We should pull out the hash type from the `FileResult` and put it as a property here instead. 

This saves a few redundant bytes per hash entry, and makes it easier to add more hashing algorithms, as there is an entity size limit of 1MiB that we may hit if we shove all of them in the same entity. 

Then we should key these documents by repo + hash type + git ref.